### ISSUE TITLE
Use Webpack to bundle your app

### ISSUE THREAD

**Issue by kddrake** [Jropp/jason-ropp-js-iobp#26]
# Dependencies
## Please complete these issues before attempting this one
- [ ] #20

## Main Goal
Rather than serving _all_ of your static files individually, bundle'm up with Webpack and the bundle(s) instead.

### When Accomplished
- [x] Running `npm start` will, in addition to everything it currently does, bundles your project into `dist/`
- [x] The server hosts `dist/`
- [x] Browser works as it did before
- [ ] Images bundle properly

## Notes
This section is a new one and one I'm super knowledgeable on. I'd highly suggest really digging through the [Webpack documentation](url). Javascript files may not be the only things you want to bundle. That'll be up to you.

# Project Notes

### HTML Files
Html files also had to be copied over

### CSS Files
CSS files also had to be copied over

#### Solution
[This article helped me walk through it quickly](https://medium.com/a-beginners-guide-for-webpack-2/webpack-loaders-css-and-sass-2cc0079b5b3a)

1. `npm i css-loader style-loader --save-dev`
2. webpackconfig settings in module.rules

```
{
  test: /\.css$/,
  use: ['style-loader', 'css-loader']
}
```
3. `import "path/to/main.css;" <--or any of your other css files. Default js for webpack 4 is index.js
Babel-loader wasn't working correctly in .webpackconfig but did when I put it in the script to run webpack 

## Project Changes

**Issues order**

js bundled
html bundled -links to html-webpack-plugin
css bundled

**Comment by Jropp** [comment]
## Changes to Webpack after Collin Walkthrough

```
plugins: [
		new CleanWebpackPlugin('dist'),
		new CopyWebpackPlugin([
			{ from: 'src/components/*', to: '[name].[ext]' },
			{ from: 'src/pages/*', to: '[name].[ext]' },
			{ from: 'src/index.html', to: '[name].[ext]' },
			{ from: 'node_modules/@banno/polymer/polymer.html', to: '[name].[ext]' },
			{ from: 'src/styles.css', to: '[name].[ext]' },
		]),
		// Does the same thing as copywebpackplugin but for webpack-dev-server
		new WriteFilePlugin(),
	],

	// Anything that wp doesn't generate on its own gets served out of dist
	devServer: {
		contentBase: 'dist'
	}
```

**Comment by kddrake** [comment]
closed in #35 or #36
